src/yaml/aslYamlLanguageService.ts

```diff
@@ -0,0 +1,228 @@
+                position.character -= 1
+            }
+        }
+        if (currentDoc) {
+            const aslCompletions : CompletionList  = doCompleteAsl(document, position, currentDoc, yamlCompletions, {
+                ignoreColonOffset: true,
+            })
+
+            aslCompletions.items.forEach(completion => {
+                // format json completions for yaml
+                if (completion.textEdit) {
+                    // textEdit can't be done on white-space so insert text is used instead
+                    if (atSpace) {
+                        completion.insertText = completion.textEdit.newText
+                        // remove any commas from json-completions
+                        completion.insertText = completion.textEdit.newText.replace(/[\,]/g, '')
+                        completion.textEdit = undefined
+                    } else {
+                        completion.textEdit.range.start.character = position.character
+                    }
+                }
+            })
+
+            return Promise.resolve(aslCompletions)
+        } else {
+            return { items: [], isIncomplete: false }
+        }
+    }
+
+    languageService.doHover = function(
+        document: TextDocument,
+        position: Position,
+        jsonDocument: JSONDocument
+    ): Thenable<Hover | null> {
+        const doc = parseYAML(document.getText())
+        const offset = document.offsetAt(position)
+        const currentDoc = matchOffsetToDocument(offset, doc)
+        if (!currentDoc) {
+            // tslint:disable-next-line: no-null-keyword
+            return Promise.resolve(null)
+        }
+
+        const currentDocIndex = doc.documents.indexOf(currentDoc)
+        currentDoc.currentDocIndex = currentDocIndex
+
+        return aslLanguageService.doHover(document, position, currentDoc)
+    }
+
+    languageService.format = function(document: TextDocument, range: Range, options: FormattingOptions): TextEdit[] {
+        try {
+            const text = document.getText()
+
+            const formatted = prettier.format(text, { parser: 'yaml' })
+
+            return [TextEdit.replace(Range.create(Position.create(0, 0), document.positionAt(text.length)), formatted)]
+        } catch (error) {
+            return []
+        }
+    }
+
+    languageService.findDocumentSymbols = function(document: TextDocument): SymbolInformation[] {
+        const doc = parseYAML(document.getText())
+        if (!doc || doc.documents.length === 0) {
+            return []
+        }
+
+        let results: any[] = []
+        for (const yamlDoc of doc.documents) {
+            if (yamlDoc.root) {
+                results = results.concat(aslLanguageService.findDocumentSymbols(document, yamlDoc))
+            }
+        }
+
+        return results
+    }
+
+    languageService.findDocumentSymbols2 = function(document: TextDocument): DocumentSymbol[] {
```

Comment:
What's the difference between version 1 and version 2? Would be good to have more descriptive names.

Reply:
It's part of the [LanguageService interface](https://github.com/microsoft/vscode-json-languageservice/blob/master/src/jsonLanguageService.ts#L46) defined by Microsoft...

We previously discussed that here: https://github.com/aws/aws-toolkit-vscode/pull/1255#discussion_r473282931

Reply:
Interesting, thanks for the info!